jupyter-extension/jupyterlab_pachyderm/mount_server_client.py

```diff
@@ -82,7 +92,7 @@ async def _ensure_mount_server(self):
             if not await self._is_mount_server_running():
                 self._unmount()
                 mount_server_cmd = f"mount-server --mount-dir {self.mount_dir}"
-                if self.nopriv:
+                if self.nopriv or self.determined_resources_type == SLURM_JOB:
```

Comment:
Would be cleaner to do this test in initial definition of self.nopriv instead of every reference.

Another option that I thought of overnight is that we could check the container runtime type.   Singularity/Apptatiner set variables.  enroot/podman don't set any variables.  

```
Apptainer> env |grep SINGULARITY
SINGULARITY_NAME=a247748d14f7b0f71058b0eb393d8d583ae3f328836bd99c386ca7fb1e1e0482
SINGULARITY_ENVIRONMENT=/.singularity.d/env/91-environment.sh
SINGULARITY_CONTAINER=/home/users/harrow/.apptainer/cache/oci-tmp/a247748d14f7b0f71058b0eb393d8d583ae3f328836bd99c386ca7fb1e1e0482
SINGULARITY_BIND=
Apptainer>
```

Reply:
this could be restrictive for the container types, since it will always use the no_priv option. these containers could be used in non hpc environment as well.
however if need that information for some other purpose we can set some variables for the same.

Reply:
| these containers could be used in non hpc environment as well.
singularity/apptainer/enroot are root-less containers.   You never have root permissions in them (without running from the root account) and cannot generally do mounts.     I guess you could additionally check if you were running as root instead of a user -- or you could enable NONPRIV_CONTAINER=0 to disable it.